### DISCUSSION TITLE
Structured output?

### DISCUSSION THREAD

**Discussion by mattdsteele** [fastlane/fastlane#16908] [Q&A]
I'm writing a Node app that uses [childProcess.exec](https://nodejs.org/api/child_process.html#child_process_child_process_execsync_command_options) to run Fastlane calls, such as `deliver` and `supply`.

Sometimes the build is successful, but there are issues identified. Things like:

* Metadata categories are deprecated
* Screenshots fail to upload (hit max # of screenshots, etc)
* Precheck warnings/failures
* etc

Are there any ways to get these kinds of outputs in a structured manner? i.e. a JSON response with an array for unexpected behavior?

I see an [environment variable](http://docs.fastlane.tools/advanced/fastlane/#output-environment-variables) `FASTLANE_DISABLE_OUTPUT_FORMAT`, but it doesn't appear to be the same thing?

If not, I'm going to try and just parse stdout, but I want to make sure I'm not missing anything 🙃

**Comment by joshdholtz** [comment]
Hey 👋 You could maybe do something with the `error` block - http://docs.fastlane.tools/advanced/lanes/#passing-parameters

```rb

error do |lane, exception, options|
  # Write something with error to a file for node to read?
end

```

I don’t know if this will always work but its an option 🤷‍♂️

**Reply by mattdsteele** [reply]
I presume this would go in a Fastfile? We currently just pass everything as CLI flags; maybe this is the feature that pushes us over the edge :)
